perun-core/src/main/java/cz/metacentrum/perun/core/impl/MembersManagerImpl.java

```diff
@@ -912,4 +975,13 @@ private String getGroupStatusSQLConditionForMembersPage(MembersPageQuery query,
 		return groupStatusesQueryString;
 	}
 
+	private boolean isVoAdminOrObserver(PerunSession sess, Vo vo) {
+		try {
+			var query = jdbc.query("SELECT 1 FROM authz WHERE user_id=? AND vo_id=? AND (role_id=? OR role_id=?)", (rs, i) -> true,sess.getPerunPrincipal().getUserId(), vo.getId(), AuthzResolverBlImpl.getRoleIdByName(Role.VOADMIN), AuthzResolverBlImpl.getRoleIdByName(Role.VOOBSERVER));
+			return !query.isEmpty();
+		} catch (InternalErrorException e) {
+			log.error("Error during checking if user is vo admin of vo {}", vo, e);
+		}
+		return false;
+	}
```

Comment:
This probably belongs in AuthzResolver, could you please move it there? Methods like `isVoAdmin()` already exist, just need to be extended a bit.

Reply:
The method `isVoAdmin` does not check if the user is VoAdmin for concrete VO. It only checks the session, if the user has the role but nothing else.